CodenameOne/src/com/codename1/util/DateUtil.java

```diff
@@ -155,14 +348,141 @@ public boolean isSameDay(Date d1, Date d2) {
+     */
+    public boolean isSameHour(Date d1, Date d2) {
+        Calendar c1 = Calendar.getInstance(tz);
+        c1.setTime(d1);
+        Calendar c2 = Calendar.getInstance(tz);
+        c2.setTime(d2);
+
+        return isSameHour(c1, c2);
+    }
+
+    /**
+     * Checks if two times are on the same minute using {@link Calendar}.
+     *
+     * @param c1 First time to compare.
+     * @param c2 Second time to compare.
+     *
+     * @return True if the two times are on the same minute.
+     * @since 7.0
+     */
+    public boolean isSameMinute(Calendar c1, Calendar c2) {
+
+        return isSameHour(c1, c2)
+                && c1.get(Calendar.MINUTE) == c2.get(Calendar.MINUTE);
+    }
+
+    /**
+     * Checks if two dates are on the same minute in the current timezone.
+     *
+     * @param d1 First date to compare.
+     * @param d2 Second date to compare.
+     *
+     * @return True if the two dates are on the same minute.
+     * @since 7.0
+     */
+    public boolean isSameMinute(Date d1, Date d2) {
+        Calendar c1 = Calendar.getInstance(tz);
+        c1.setTime(d1);
+        Calendar c2 = Calendar.getInstance(tz);
+        c2.setTime(d2);
+
+        return isSameMinute(c1, c2);
+    }
+
+    /**
+     * Checks if two times are on the same second using {@link Calendar}.
+     *
+     * @param c1 First time to compare.
+     * @param c2 Second time to compare.
+     *
+     * @return True if the two times are on the same second.
+     * @since 7.0
+     */
+    public boolean isSameSecond(Calendar c1, Calendar c2) {
+
+        return isSameMinute(c1, c2)
+                && c1.get(Calendar.SECOND) == c2.get(Calendar.SECOND);
+    }
+
+    /**
+     * Checks if two dates are on the same second in the current timezone.
+     *
+     * @param d1 First date to compare.
+     * @param d2 Second date to compare.
+     *
+     * @return True if the two dates are on the same second.
+     * @since 7.0
+     */
+    public boolean isSameSecond(Date d1, Date d2) {
+        Calendar c1 = Calendar.getInstance(tz);
+        c1.setTime(d1);
+        Calendar c2 = Calendar.getInstance(tz);
+        c2.setTime(d2);
+
+        return isSameSecond(c1, c2);
+    }
+
+    /**
+     * Checks if two times are on the same time using {@link Calendar}.
+     *
+     * @param c1 First time to compare.
+     * @param c2 Second time to compare.
+     *
+     * @return True if the two times are on the same time.
+     * @since 7.0
+     */
+    public boolean isSameTime(Calendar c1, Calendar c2) {
+
+        return c1.getTime().getTime() == c2.getTime().getTime();
+    }
+
+    /**
+     * Checks if two dates are on the same time in the current timezone.
+     *
+     * @param d1 First date to compare.
+     * @param d2 Second date to compare.
+     *
+     * @return True if the two dates are on the same time.
+     * @since 7.0
+     */
+    public boolean isSameTime(Date d1, Date d2) {
+        Calendar c1 = Calendar.getInstance(tz);
```

Comment:
You can just avoid the calendar object for this specific case as `d1.getTime() == d2.getTime()` is fine and isn't deprecated.